src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const allProperties = sourceString.split(';');
+
+  const resultObject = allProperties.reduce((currentObject, property) => {
+    const part = property.split(':');
+    const partName = part[0]?.trim();
+    const partValue = part[1]?.trim();
+
+    if (partName && partValue) {
+      currentObject[partName] = partValue;
+    }
+
+    return currentObject;
+  }, {});
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. The `reduce` method is appropriately used here to iterate over the array and build up the result object.